### PR TITLE
feat(rspack): update withReact plugin for parity with webpack plugin

### PR DESCRIPTION
**Pull request by ndcunningham**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/rspack/src/utils/lib/apply-react-config.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        ...(babelLoader.options['plugins'] || []),
+        isDev && [
+          require.resolve('react-refresh/babel'),
+          { skipEnvCheck: true },
+        ],
+      ].filter(Boolean);
+    }
+  }
+
+  config.plugins = [
```
Comment (jaysoo) [inline packages/rspack/src/utils/lib/apply-react-config.ts:83]: Can't we just do this?

```
if (isDev) {
  config.plugins.push(new ReactRefreshPlugin({ overlay: false }));
}
```

**Review by jaysoo — APPROVED** [review]

File: packages/rspack/src/utils/with-react.ts [hunk 2]
```diff
@@ -1,77 +1,29 @@
 import { Configuration } from '@rspack/core';
-import { withWeb } from './with-web';
 import { NxRspackExecutionContext } from './config';
+import { withWeb, WithWebOptions } from './with-web';
+import { applyReactConfig } from './lib/apply-react-config';
 
-export function withReact(opts = {}) {
+export interface SvgrOptions {
```
Comment (Coly010) [inline packages/rspack/src/utils/with-react.ts:6]: You'll need to move this out of here and likely into `src/plugins/utils/models.ts`

There's a circular dep now between `with-react` and `apply-react-config`

File: packages/rspack/src/utils/lib/apply-react-config.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import { Configuration, RspackOptionsNormalized } from '@rspack/core';
+import { SvgrOptions } from '../with-react';
```
Comment (Coly010) [inline packages/rspack/src/utils/lib/apply-react-config.ts:2]: This is part 2 of the circular dep

**Review by Coly010 — CHANGES_REQUESTED** [review]
Just a circular dep removal and this should be fine.

Has it been tested with Standard React app and with Module Federation?

**Review by Coly010 — APPROVED** [review]

**Pull request merged**